### PR TITLE
Updates to improve versatility in implementation

### LINKED ISSUE #199: Remove `select__label` and `select__control` classes from Select component
## Describe the bug
The `select__label` and `select__control` classes which are included in the Select component do not have corresponding styles, so they can be removed.

## Expected behavior
Classes should either have styles associated with them, or should not exist.

## Impact on your work
Minor bug, but good to get resolved before wider implementation of the RAC style library.

### LINKED ISSUE #200: Improve accessibility of Pagination component
## Is your feature request related to a problem? Please describe.
The accessibility of the Pagination component can be improved.

## Describe the solution you'd like
- Remove `tabindex` values
- remove `role="button"` attributes
- Update the truncated pages button:
  - add the `disabled` class and `aria-disabled=true` to the link.
  - add an `aria-label` attribute that says "truncated pages"

## Additional context
The pagination on Argo may be a good pattern to refer to, since that has received recent attention.

### LINKED ISSUE #203: Update form validation patterns
## Describe the bug
We have implemented required form elements and validation/error messages inconsistently across different form components. Some, like `search`, use client-side browser validation and `aria-required` is inconsistently applied. Also, we need to research if `aria-invalid="false"` should be set for required fields that are valid (I've seen conflicting recommendations).

## Expected behavior
When fields are required or invalid, that should be clearly and consistently communicated to screen reader users.

## Additional context
[A useful breakdown of a11y form basics](https://www.tpgi.com/required-attribute-requirements/), and also this [Exposing Field Errors](https://adrianroselli.com/2023/04/exposing-field-errors.html) post

### LINKED ISSUE #204: Adjust h2 and input error styles, and add font bold helper class
## Is your feature request related to a problem? Please describe.
From Style Library implementation in aquila ([PR #303](https://github.com/RockefellerArchiveCenter/aquila/pull/303)) and argo ([PR #281](https://github.com/RockefellerArchiveCenter/argo/pull/281)), some additional helper classes and adjustments would be useful to have to add the same local custom styles in multiple other projects.

## Describe the solution you'd like
- Add a `font-weight--bold` class that just makes text bold to typography
- Adjust `input__error` styles to have a width of 100% so that they stay in there own line in a flex layout
- H2 has padding below it that makes way too much space. This is because our original h2 from rockarch.org had a dotted border below it, and the padding is for that. Make that padding style part of the “heading—dotted-border" class in the style library, not the h2.
- Add a `flex--column` class with `flex-direction: column`.

### LINKED ISSUE #205: Make card component more flexible
## Is your feature request related to a problem? Please describe.
From Style Library implementation in aquila (https://github.com/RockefellerArchiveCenter/aquila/pull/303) and argo (https://github.com/RockefellerArchiveCenter/argo/pull/281), we found that using the card component was useful beyond just clickable lists of items. But to make those styles work, we had to apply local custom styles to undo Style Library styles that make the whole card focusable/clickable.

## Describe the solution you'd like
Make the card styles more flexible. In the RAC Style Guide/Library, they were conceived to be a list of things where the card itself is clickable. Expand these to be more flexible as containers of other kinds of content. This will likely mean new card classes.

### LINKED ISSUE #206: Remove unnecessary tabindex values from header menu
## Is your feature request related to a problem? Please describe.
In the header dropdown menu for mobile, the Style Guide HTML includes tabindex values. These were carried over from the DIMES HTML, and specific to the React component we used in that project. The are not needed in the canonical header dropdown menu.

## Describe the solution you'd like
Remove the unnecessary tabindex values. Review other components for similar unnecessary script-related attributes. Where necessary, improve documentation for HTML attributes that are specific to scripts.

### PR DESCRIPTION
**Pull request by HaSistrunk**
Closes #199 , #200 ,  #204 , #205 , #206 , #203  
   
Updates the Style Library and Guide identified as necessary from aquila and argo styles implementation. Documenting the changes in detail here to make implementation back into sites easier.

Global changes (fixes #204 ):
- Change `link--bold` class to `text--bold` for wider application
- Give `input__error` class a width of 100%
- Add a new helper class `flex--column` to apply `flex-direction: column` styles to a flex container
- Move bottom padding from h2 to `heading--dotted-border` class.

Form improvements (fixes #199 ):
- Groups form components in Style Guide
- Remove hidden labels from basic text inputs (better UX and a11y to avoid hiding labels)
- Add basic custom styles for select, input, and textinput styles
- Increase form error message font to 14px
- Remove unused classes from inputs
- Remove `aria-required="true"` from the search component. We only need `required` like the other inputs have (fixes #203  )

Make cards more flexible (fixes #205 ):
- Add a `card--container` class to apply to cards that aren't clickable to change their hover/focus styles.
- Remove `list--unstyled` from card HTML and add to card scss instead.

Improve header dropdown (resolve #206 ):
- Add z-index to mobile nav dropdown to keep other page content from covering it.
- Remove `tabindex="0"` from nav mobile dropdown button and add ids for scripts (`dropdown__list--mobile`). We can take a closer look at this when we standardize our scripts in the style library.

Simplify pagination (fixes #200 ):
- Remove unused attributes and improve accessibility

### CODE REVIEW

File: stories/components/card/Card.mdx [hunk 1]
```diff
@@ -41,10 +41,20 @@ For example, for 4 card columns on large screens and 3 columns on medium screens
...
+By default, cards are designed to be a list of links with a body and a footer.
 
 <Story of={stories.basic} />
 
 However, the card footer is optional.
 
 <Story of={stories.noFooter} />
+
+## Using cards as stylized content containers
+A card can be used flexibly as a content container that is not
```
Comment (helrond) [inline stories/components/card/Card.mdx:53]: Perhaps I am parsing this too finely, but it seems to me that if the default behavior is that a card is a link then maybe we should be _adding_ a class to make it not a link?

File: stories/components/card/card.handlebars [hunk 2]
```diff
@@ -1,13 +1,27 @@
-<ul class="card-list list--unstyled">
-  <li class="card">
-    <div class="card__body">
-      <a class="card__title" href="{{linkTarget}}">{{titleText}}</a>
-      <p class="card__body-text">{{bodyText}}</p>
+{{#if link}}
+  <ul class="card-list list--unstyled">
```
Comment (helrond) [inline stories/components/card/card.handlebars]: I'm just noticing this now, but can we incorporate the `list--unstyled` rules into the `card-list` rules so we don't have to add two rules here?

**Review by helrond — CHANGES_REQUESTED** [review]
A couple of minor things on the card element.

File: stories/components/card/Card.mdx (hunk 1 above)
Comment (HaSistrunk) [reply stories/components/card/Card.mdx:53]: Good point. Updating this to add a `.card--container` class instead.

**Review by helrond — APPROVED** [review]

**Review by p-galligan — APPROVED** [review]

**Pull request merged**